core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -331,7 +331,8 @@ public HttpResponse doConfigSubmit(StaplerRequest req, StaplerResponse rsp) thro
         j.clouds.replace(this, result);
         j.save();
         // take the user back to the cloud top page.
-        return FormApply.success(".");
+        return FormApply.success("../" + result.name + '/');
```

Comment:
for what it's worth, I think return FormApply.success("../.."); is the quick fix for the issue. Due to the above underlying url. You will always be in .../configure so you are forced to get back to the overall cloud listing page. If you want some elegance you test whether the name has not changed, then use the old way. However, since index.jelly is e.g. in kubernetes plugin a blank page and doing "." leads to it I would rather be redirected to the overall cloud page, but that is me as a user.

Reply:
@scherler I think the main issue with that was that a quick fix was not desired, but rather a proper way to address the UX behavior
See https://github.com/jenkinsci/jenkins/pull/8310#discussion_r1292331498